### PR TITLE
Adds node 4.0.0 compatibility, updates and migrates to nan 2

### PR DESCRIPTION
**Pull request by KenanSulayman**
Yo,

I was working with https://github.com/dazoe/ed25519 and found it being too much of a liability to maintain. Then I found out about sodium and migrated it so that it works with 4.0.0.

All tests pass.

Cheers and thanks for sodium

### CODE REVIEW

File: lib/toBuffer.js [hunk 1]
```diff
@@ -49,7 +49,8 @@ function toBuffer(value, encoding) {
             }
         }
     }
+
     throw new Error('[toBuffer] unsupported type in value. Use Buffer, string or Array');
 }
 
-module.exports = toBuffer;
\ No newline at end of file
+module.exports = toBuffer;
```
Comment (paixaop) [inline lib/toBuffer.js]: Since there are no changes to the toBuffer file can you remove it from the pull request?

File: sodium.cc [hunk 2]
```diff
@@ -23,34 +23,34 @@ using namespace v8;
...
 
-// Retrieve the buffer constructor function 
-Local<Function> bufferConstructor = 
-       Local<Function>::Cast(globalObj->Get(NanNew<String>("Buffer")));
-       
 
 // Check if a function argument is a node Buffer. If not throw V8 exception
 #define ARG_IS_BUFFER(i,msg) \
-    if (!Buffer::HasInstance(args[i])) { \
+    if (!Buffer::HasInstance(info[i])) { \
```
Comment (paixaop) [inline sodium.cc]: Why change args for info? Is this just cosmetic?

Comment (KenanSulayman) [inline sodium.cc]: In nan 2.0, `NAN_METHOD` expands to a method signature with an argument list called `info` (as opposed to args in nan 1). See https://github.com/nodejs/nan/blob/f14e04a3492ede8ca2aac8f049a60c595d7bfbf2/nan.h#L1266 for further information.

File: lib/toBuffer.js (hunk 1 above)
Comment (KenanSulayman) [inline lib/toBuffer.js]: Sure.

**Comment by KenanSulayman** [comment]
@paixaop please check the updated diff.

**Comment by paixaop** [comment]
Thanks!!!

**Comment by KenanSulayman** [comment]
@paixaop sure! Could you please release it?

**Comment by paixaop** [comment]
Done. sorry it took me a few days to get around to it.

**Comment by KenanSulayman** [comment]
Awesome, thanks @paixaop!

**Pull request merged**